sys/syz-sysgen/check.go

```diff
@@ -0,0 +1,30 @@
+// Copyright 2023 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package main
+
+import (
+	"fmt"
+
+	"github.com/google/syzkaller/pkg/ast"
+	"github.com/google/syzkaller/pkg/compiler"
+)
+
+// constsAreAllDefined() ensures that for every const there's at least one arch that defines it.
+func constsAreAllDefined(consts *compiler.ConstFile, constInfo map[string]*compiler.ConstInfo,
+	eh ast.ErrorHandler) {
+	// We cannot perform this check inside pkg/compiler because it's
+	// given a const slice for only one architecture.
+	for _, info := range constInfo {
+		if info.NoExtract {
```

Comment:
Do we have any existing warning in noextract files? They still should have const value for at least 1 arch. Can we clean up existing warning and enable this?

Reply:
We do have them in the `sys/linux/dev_video4linux.txt` file
```
sys/linux/dev_video4linux.txt:1:1: V4L2_COLORSPACE_ADOBERGB is defined for none of the arches
sys/linux/dev_video4linux.txt:1:1: V4L2_DEC_QCOM_CMD_RECONFIG_HINT is defined for none of the arches
sys/linux/dev_video4linux.txt:1:1: V4L2_EVENT_MSM_VIDC_FLUSH_DONE is defined for none of the arches
< ... >
```

Maybe just drop all its descriptions/consts that are not present upstream?

Reply:
If most of this is upstream, then, yes, I guess we can just update them to upstream kernel and remove "noextract".

Otherwise, I would just remove consts that are not defined and all thier uses, or define them to 0 and add a comment.
This would allow to not have additional logic in sysgen and will allow to catch new bugs in "noextract" descriptions (any of these new warnings is equally a bug for "noextract" descirptions as well).

Reply:
I've dropped obsolete consts from there and regenerated it on the latest mainline. Now the `noextract` check is not necessary.